### PR TITLE
Use npm workspaces for packages (second attempt)

### PR DESCRIPTION
**Pull request by sirreal**
Update package management to use npm workspaces. Originally in https://github.com/WordPress/gutenberg/pull/65681, reverted for timing reasons.

#66314 should have fixed an issue on CI observed while the original PR was merged to trunk.

From 4693c0f9942f11f48e0912db378b1b9e2fbf4af0:

> Update package management to use npm workspaces instead of lerna.
> 
> Lerna is obsolete for many repository management operations, such as
> dependency installation or script running. All of these are supported by
> npm now.
> 
> Lerna continues to be used for publishing flows to ensure the
> appropriate dependencies are updated and published with the correct
> versions.
> 
> The hope is that this simplfies dependendy management and maintains a
> more sane dependency tree. Dependency management has become increasingly
> difficult as Gutenberg has grown.
> 
> ---
> 
> Co-authored-by: t-hamano <wildworks@git.wordpress.org>
> Co-authored-by: sirreal <jonsurrell@git.wordpress.org>
> Co-authored-by: gziolo <gziolo@git.wordpress.org>
> Co-authored-by: ciampo <mciampini@git.wordpress.org>
> Co-authored-by: jsnajdr <jsnajdr@git.wordpress.org>
> Co-authored-by: yusuke-omae <omaeyusuke@git.wordpress.org>
> Co-authored-by: kevin940726 <kevin940726@git.wordpress.org>

### CODE REVIEW

**Comment by gziolo** [comment]
Revert of the revert 😄 Let me know when to approve it again.

**Comment by sirreal** [comment]
There was a strange failure on CI while this was on trunk: https://github.com/WordPress/gutenberg/actions/runs/11435991643/job/31813748278

**Comment by sirreal** [comment]
[In the pull-request-automation workflow](https://github.com/WordPress/gutenberg/blob/trunk/.github/workflows/pull-request-automation.yml) we hit this issue:

https://github.com/ds300/patch-package/issues/277

- Patch package is not installed, so the repo-wide `postinstall` script fails.
- It can be installed, but then it fails because some patches cannot be applied because they're not installed.

I think we should just use a single `npm ci` of the entire repo (that's likely cached on trunk) instead of a separate cache.

**Comment by sirreal** [comment]
There's an unrelated static analysis failure currently on trunk being fixed in https://github.com/WordPress/gutenberg/pull/66316.

You can see pull-request-automation passing on this branch here: https://github.com/WordPress/gutenberg/actions/runs/11459079724/job/31882816645?pr=66272 (this was a problem on trunk when this landed before).

File: packages/vips/package.json [hunk 1]
```diff
@@ -31,5 +31,6 @@
 	},
 	"publishConfig": {
 		"access": "public"
-	}
+	},
+	"wpScript": true
```
Comment (sirreal) [inline packages/vips/package.json:35]: This was not in the original PR, this package was added in the interim in #64845.

**Review by gziolo — APPROVED** [review]
Let's try again 🚢

**Comment by sirreal** [comment]
Linking interesting follow-up conversation: https://github.com/WordPress/gutenberg/pull/66428#issuecomment-2437498575

**Pull request merged**